rest-api/api/metrics_api.py

```diff
@@ -31,7 +32,12 @@ def post(self):
         try:
           end_date = datetime.datetime.strptime(end_date_str, DATE_FORMAT).date()
         except ValueError:
-          raise BadRequest("Invalid start date: %s" % end_date_str)
+          raise BadRequest("Invalid end date: %s" % end_date_str)
+      if start_date and end_date:
```

Comment:
I think we need to start requiring start_date and end_date to be set as well.

The underlying API doesn't require them, but if you don't specify them it can hose our server.

(We should confirm that HPRO never queries for metrics without specifying both start and end date.)

Reply:
In line 20 & 21 we are setting a default value None for both start_date and end_date variables so I think there is no need to check if the variable is set or not.

Reply:
My point is if you pass None into the DAO before, it won't set an upper or lower bound on the metrics we're fetching... which can lead to more than 7 days of results, and memory exploding. We shouldn't allow that.

Reply:
I think setting `buckets = None` by default and placing `buckets = dao.get_active_buckets(start_date, end_date)` in side `if start_date and end_date:` condition should solve this issue.

Reply:
It doesn't make sense to return None.

If start_date and end_date are required, we should throw a BadRequest / return 400 if you don't specify them... like you are doing here if they are too far apart. Does that make sense?